### PR TITLE
New Resource: `azurerm_cost_management_export_resource_group`

### PR DESCRIPTION
**Pull request by mbfrahry**
```
--- PASS: TestAccAzureRMCostManagementExportResourceGroup_basic (147.15s)
--- PASS: TestAccAzureRMCostManagementExportResourceGroup_update (175.36s)
```

### CODE REVIEW

File: azurerm/internal/services/costmanagement/resource_arm_cost_management_export_resource_group.go [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+	properties := &costmanagement.ExportProperties{
+		Schedule:     schedule,
+		DeliveryInfo: expandExportDeliveryInfo(d.Get("delivery_info").([]interface{})),
+		Format:       costmanagement.Csv,
+		Definition:   expandExportQuery(d.Get("query").([]interface{})),
+	}
+
+	account := costmanagement.Export{
+		ExportProperties: properties,
+	}
```
Comment (katbyte) [inline azurerm/internal/services/costmanagement/resource_arm_cost_management_export_resource_group.go]: Should we consolidate this to be consistent with other resources?
```suggestion
	schedule := 

	account := costmanagement.Export{
		ExportProperties: &costmanagement.ExportProperties{
			Schedule:     &costmanagement.ExportSchedule{
				Recurrence: costmanagement.RecurrenceType(d.Get("recurrence_type").(string)),
				RecurrencePeriod: &costmanagement.ExportRecurrencePeriod{
					From: &date.Time{Time: from},
					To:   &date.Time{Time: to},
				},
				Status: status,
			},
			DeliveryInfo: expandExportDeliveryInfo(d.Get("delivery_info").([]interface{})),
			Format:       costmanagement.Csv,
			Definition:   expandExportQuery(d.Get("query").([]interface{})),
		},
	}
```

File: azurerm/internal/services/costmanagement/resource_arm_cost_management_export_resource_group.go [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+		return fmt.Errorf("retrieving Cost Management Export Resource Group %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("cannot read Cost Management Export Resource Group %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	id := *resp.ID
+	// The ID is missing the prefix `/` which causes our uri parse to fail
+	if !strings.HasPrefix(*resp.ID, "/") {
```
Comment (katbyte) [inline azurerm/internal/services/costmanagement/resource_arm_cost_management_export_resource_group.go]:
```suggestion
	if !strings.HasPrefix(id, "/") {
```

File: azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMCostManagementExportResourceGroup_update(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCostManagementExportResourceGroupExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
```
Comment (katbyte) [inline azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go]: Should we add a basic step here?

File: azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+func testAccAzureRMCostManagementExportResourceGroup_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-cmerg-%d"
```
Comment (katbyte) [inline azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go]: We can just put the service here
```suggestion
  name     = "acctestRG-cm-%d"
```

File: azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+  name                = "unlikely23exst2acct%s"
+  resource_group_name = azurerm_resource_group.test.name
+
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_cost_management_export_resource_group" "test" {
+  name                    = "accrg%d"
```
Comment (katbyte) [inline azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go]: Should we use some capitals here?

**Review by katbyte — APPROVED** [review]
Aside from a couple comments LGTM 👍

File: azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go (hunk 5 above)
Comment (mbfrahry) [reply azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go]: Not this time. It only accepts lower case

**Pull request merged**